sections/exchange/hooks/useExchange.tsx

```diff
@@ -179,18 +191,27 @@ const useExchange = ({
 	const [gasLimit, setGasLimit] = useState<number | null>(null);
 
 	const { base: baseCurrencyKey, quote: quoteCurrencyKey } = currencyPair;
-	const ETHBalanceQuery = useETHBalanceQuery();
+	const ETHBalanceQuery = useETHBalanceQuery(walletAddress);
 	const ETHBalance = ETHBalanceQuery.isSuccess ? ETHBalanceQuery.data ?? zeroBN : null;
 
-	const synthsWalletBalancesQuery = useSynthsBalancesQuery();
+	const synthsWalletBalancesQuery = useSynthsBalancesQuery(walletAddress);
 	const synthsWalletBalance = synthsWalletBalancesQuery.isSuccess
 		? synthsWalletBalancesQuery.data
 		: null;
 
 	const ethGasPriceQuery = useEthGasPriceQuery();
 	const exchangeRatesQuery = useExchangeRatesQuery();
-	const feeReclaimPeriodQuery = useFeeReclaimPeriodQuery(quoteCurrencyKey);
-	const exchangeFeeRateQuery = useExchangeFeeRate(quoteCurrencyKey, baseCurrencyKey);
+
+	// TODO: these queries break when `txProvider` is not `synthetix` and should not be called.
+	// however, condition would break rule of hooks here
+	const feeReclaimPeriodQuery = useFeeReclaimPeriodQuery(
```

Comment:
`if (synthetix)` breaks rules of hooks right? becuase if `synthetix` starts uninitialized, then becomes initialized while the component is mounted, that would break the rules of hooks, right?

Reply:
Yeah you can't use conditionals, but you can tell react-query not to query if one of the conditions are false. 

https://react-query.tanstack.com/guides/disabling-queries

So here (Kwenta) you would do:

```
const feeReclaimPeriodQuery = useFeeReclaimPeriodQuery(
    quoteCurrencyKey as CurrencyKey,
    walletAddress,
    { enabled: txProvider === 'Synthetix' } // options
);
```

And here (queries lib) a little more tricky but you'd probably have to `...options` spread first then override with `enabled: CURRENT_LOGIC && (options.enabled ?? true)`

https://github.com/Synthetixio/js-monorepo/blob/master/packages/queries/src/queries/synths/useFeeReclaimPeriodQuery.ts#L25